sky/provision/aws/instance.py

```diff
@@ -46,24 +46,15 @@
 # https://aws.amazon.com/ec2/pricing/on-demand/#Data_Transfer_within_the_same_AWS_Region
 
 
-@aws.import_package
-def botocore_config():
-    """Get botocore.config module.
-
-    This is for backward compatibility when `aws` module is imported before the
-    `botocore_config()` function is added in #1702.
-    TODO: remove this function after 0.6.0 release, when new provisioner is
-    used by default.
-    """
-    config = aws.botocore.config
-    return config
-
-
 def _default_ec2_resource(region: str) -> Any:
+    if not hasattr(aws, 'version'):
+        # For backward compatibility, reload the module if the aws module was
+        # imported before and staled.
```

Comment:
```suggestion
        # imported before and stale. Used for, e.g., a live spot controller running an older version and a new version gets installed by `sky spot launch`.
```
Q: actually, this reload() won't affect any existing controller.py processes? A bit confused why we need back compat here.

Reply:
Previously, the spot controller’s backward compatibility is guaranteed by the nature that python will cache all the imported module in the memory, so even if the skypilot package is updated in the controller VM, the existing controller process will still use the old code, i.e., no backward compatibility issue.
However, since we are using this `adaptor` style implementation, the modules are only imported when used, this causes any update in the `aws.instance` will be used by the old controller process. That said, the `adaptors.aws` is in old version while the `aws.instance` is in the new version, causing a mismatch in the implementation.

Reply:
This is only a temporary solution, and I think we should get rid of the `adaptors` but only use the `provision` APIs as the two-level import and reloading can cause a lot of issues, like the older code request older `adaptor.aws` while the newly imported `aws.instance` requires newer `adaptor.aws`.